glue/clients/histogram_client.py

```diff
@@ -229,10 +229,10 @@ def _auto_nbin(self, calculate_only=False):
         c = list(self._get_data_components('x'))
         if c:
             c = c[0]
-            if isinstance(c, CategoricalComponent):
-                val = min(c._categories.size, 100)
+            if c.categorical:
```

Comment:
This is fine for now. As discussed in other threads, this kind of logic branch on component type information should probably instead be implemented as category methods, so that histogram clients don't need to know as much about how components behave. But I think @JudoWill's PR will address that